ipatests/test_integration/test_sssd.py

```diff
@@ -53,58 +50,105 @@ def install(cls, mh):
 
         cls.users['ad']['name'] = cls.users['ad']['name_tmpl'].format(
             domain=cls.ad.domain.name)
+        cls.users['ad']['group'] = cls.users['ad']['group_tmpl'].format(
+            domain=cls.ad.domain.name)
         tasks.user_add(cls.master, cls.intermed_user)
         tasks.create_active_user(cls.master, cls.ipa_user,
                                  cls.ipa_user_password)
 
     @contextmanager
-    def config_sssd_cache_auth(self, cached_auth_timeout):
+    def config_sssd(self, opts):
         sssd_conf_backup = tasks.FileBackup(self.master, paths.SSSD_CONF)
         with tasks.remote_ini_file(self.master, paths.SSSD_CONF) as sssd_conf:
-            domain_section = 'domain/{}'.format(self.master.domain.name)
-            if cached_auth_timeout is None:
-                sssd_conf.remove_option(domain_section, 'cached_auth_timeout')
-            else:
-                sssd_conf.set(domain_section, 'cached_auth_timeout',
-                              cached_auth_timeout)
-            sssd_conf.set('pam', 'pam_verbosity', '2')
-
+            for section, option, value in opts:
+                if hasattr(section, 'name'):
+                    section = 'domain/{}'.format(section.name)
+                if value is None:
+                    sssd_conf.remove_option(section, option)
+                else:
+                    sssd_conf.set(section, option, value)
         try:
             tasks.clear_sssd_cache(self.master)
             yield
         finally:
             sssd_conf_backup.restore()
             tasks.clear_sssd_cache(self.master)
 
+    # Check user authentication is cached for domain and its subdomains
+    # Regression tests for https://bugzilla.redhat.com/show_bug.cgi?id=1685581
+
```

Comment:
Nitpick but the comment should be a doc string inside the method.

Reply:
well, there are three testcases + one utility for this bug, I wanted to avoid copy-pasting comments by "splitting" test-class with this comment. But it seems it is not that obvious.